aws/resource_aws_emr_managed_scaling_policy.go

```diff
@@ -0,0 +1,147 @@
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"compute_limits": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Elem:     computeLimitsSchema(),
+			},
+		},
+	}
+}
+func computeLimitsSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"unit_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAwsEmrComputeLimitsUnitType(),
+			},
+			"minimum_capacity_units": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Default:  1,
+			},
+			"maximum_capacity_units": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Default:  1,
+			},
+			"maximum_core_capacity_units": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Default:  1,
+			},
+			"maximum_ondemand_capacity_units": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Default:  1,
+			},
+		},
+	}
+}
+
+func resourceAwsEMRManagedScalingPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).emrconn
+
+	if l := d.Get("compute_limits").(*schema.Set).List(); len(l) > 0 && l[0] != nil {
+		cl := l[0].(map[string]interface{})
+		computeLimits := &emr.ComputeLimits{
+			UnitType:                     aws.String(cl["unit_type"].(string)),
+			MinimumCapacityUnits:         aws.Int64(int64(cl["minimum_capacity_units"].(int))),
+			MaximumCapacityUnits:         aws.Int64(int64(cl["maximum_capacity_units"].(int))),
+			MaximumCoreCapacityUnits:     aws.Int64(int64(cl["maximum_core_capacity_units"].(int))),
+			MaximumOnDemandCapacityUnits: aws.Int64(int64(cl["maximum_ondemand_capacity_units"].(int))),
+		}
+		managedScalingPolicy := &emr.ManagedScalingPolicy{
+			ComputeLimits: computeLimits,
+		}
+
+		_, err := conn.PutManagedScalingPolicy(&emr.PutManagedScalingPolicyInput{
+			ClusterId:            aws.String(d.Get("cluster_id").(string)),
+			ManagedScalingPolicy: managedScalingPolicy,
+		})
+
+		if err != nil {
+			log.Printf("[ERROR] EMR.PutManagedScalingPolicy %s", err)
+			return err
+		}
+	}
+
+	d.SetId(d.Get("cluster_id").(string))
+	return nil
+}
+
+func resourceAwsEMRManagedScalingPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).emrconn
+	resp, err := conn.GetManagedScalingPolicy(&emr.GetManagedScalingPolicyInput{
+		ClusterId: aws.String(d.Get("cluster_id").(string)),
+	})
+	if err != nil {
+		if isAWSErr(err, "InvalidRequestException", "does not exist") {
+			log.Printf("[WARN] EMR Managed Scaling Policy (%s) not found, removing from state", d.Get("cluster_id").(string))
+			d.SetId("")
+			return nil
+		}
+		return err
```

Comment:
Similarly here with returning error context:

```suggestion
		return fmt.Errorf("error getting EMR Managed Scaling Policy (%s): %w", d.Id(), err)
```